diff --git a/src/tmodule.ml b/src/tmodule.ml
--- a/src/tmodule.ml
+++ b/src/tmodule.ml
@@ -614,3 +614,3 @@
   in
-  let oc = open_out filename in
+  let oc = open_out_bin filename in
   Marshal.to_channel oc md [];
@@ -619,5 +619,5 @@
 let read_gospel_file filename =
   assert (Filename.extension filename = ".gospel");
-  let ic = open_in filename in
+  let ic = open_in_bin filename in
   let md : module_uc = Marshal.from_channel ic in
   close_in ic;
