diff --git a/src/tmodule.ml b/src/tmodule.ml
--- a/src/tmodule.ml
+++ b/src/tmodule.ml
@@ -611,13 +611,13 @@ let print_file fmt { fl_nm; fl_sigs; fl_export } =
 let write_gospel_file md =
   let filename =
     Filename.chop_extension (Fmt.str "%a" Ident.pp md.muc_nm) ^ ".gospel"
   in
-  let oc = open_out filename in
+  let oc = open_out_bin filename in
   Marshal.to_channel oc md [];
   close_out oc

 let read_gospel_file filename =
   assert (Filename.extension filename = ".gospel");
-  let ic = open_in filename in
+  let ic = open_in_bin filename in
   let md : module_uc = Marshal.from_channel ic in
   close_in ic;
   md
